src/realm/db.hpp

```diff
@@ -326,16 +327,22 @@ class DB : public std::enable_shared_from_this<DB> {
     /// The name of the temporary file is formed by appending
     /// ".tmp_compaction_space" to the name of the database
     ///
-    /// If the output_encryption_key is `none` then the file's existing key will
-    /// be used (if any). If the output_encryption_key is nullptr, the resulting
+    /// If the output_encryption_key is none, the resulting
     /// file will be unencrypted. Any other value will change the encryption of
     /// the file to the new 64 byte key.
     ///
     /// WARNING: Compact() is not thread-safe with respect to a concurrent close()
-    bool compact(bool bump_version_number = false, util::Optional<const char*> output_encryption_key = util::none)
+    bool compact(bool bump_version_number, const std::optional<util::EncryptionKey>& output_encryption_key)
```

Comment:
This is an API change that should be documented in the Changelog. Alternatively, the non-breaking equivalent would be to use `const std::optional<std::optional<util::EncryptionKey>> output_encryption_key = util::none` but that is admittedly a confusing type.

Reply:
I believe the added compact overload covers usages of the old signature with the default value.